unittests/test_parameters.py

```diff
@@ -136,15 +125,13 @@ def test_extended_params_are_set():
     class MyTest(ExtendParams):
         pass
 
-    MyTest.prepare_param_space()
-    for i in range(MyTest.param_space_len()):
+    for i in range(len(MyTest.param_space)):
```

Comment:
I think this is equivalent and simpler:

```suggestion
    for _ in MyTest.param_space:
```

Reply:
With this implementation, doing that loop as `for _ in MyTest.param_space` doesn't work, because the internal iterator in the `ParamSpace` class is consumed in the wrong place and then the parameters are not initialised correctly (a `StopIteration` exception is raised during the regression test instantiation and the parameters become `None` instead of getting their right values).  This shows that the implementation for setting the parameter space is not yet very clean (or at least not very safe). I've got an idea on how to get this done in a much cleaner way. I'll try it out and then summarise it in here if successful.